lib/v4/errorMap/errorMap.ts

```diff
@@ -34,41 +30,11 @@ const issueParsers: Record<
   not_multiple_of: parseNotMultipleOfIssue,
   unrecognized_keys: parseUnrecognizedKeysIssue,
   invalid_key: parseInvalidKeyIssue,
-  custom: parseCustomIssue,
-  invalid_union: parseInvalidUnionIssue,
+  custom: parseInvalidUnionIssue,
+  invalid_union: parseCustomIssue,
 };
 
-function parseInvalidUnionIssue(
-  issue: zod.$ZodIssueInvalidUnion,
-  options: ErrorMapOptions
-): AbstractSyntaxTree {
-  const errorMap = createErrorMap(options);
-
-  const individualMessages = issue.errors.map((issues) =>
-    issues
-      .map((subIssue) =>
-        errorMap({
-          ...subIssue,
-          path: issue.path.concat(subIssue.path),
-        })
-      )
-      .join(options.issueSeparator)
-  );
-  const message = Array.from(new Set(individualMessages)).join(
-    options.unionSeparator
-  );
-
-  return {
-    type: issue.code,
-    path: [],
-    message,
-  };
-}
```

Comment:
dev: the code has been significantly simplified - the `errorMap` does not need to account for unions anymore as they are handled by the `MessageBuilder`